### PR TITLE
Remove search button, update styling

### PR DESCRIPTION
**Pull request by lucyhutcheson**
**Related Issue(s):**

- #31 Fix calendar z-index

**Proposed Changes:**

1. Remove search button
2. Perform search after filters change, on zoom end or on drag end

**PR Checklist:**

- [X] I have added my changes to the [CHANGELOG](https://github.com/Element84/filmdrop-ui/blob/main/CHANGELOG.md) **or** a CHANGELOG entry is not required.

### CODE REVIEW

File: src/components/Search/Search.js [hunk 1]
```diff
@@ -223,46 +246,58 @@ const Search = () => {
...
+  // search throttle set to 1000ms
+  const processSearch = debounce(function (resultFootprintsInit) {
+    searchAPI(resultFootprintsInit)
+  }, 1000)
+
+  // function called when search is initiated
+  function searchAPI(resultFootprintsInit) {
+    // set up footprints layer
+    let resultFootPrintsLocal = resultFootprints
+    if (resultFootprintsInit) resultFootPrintsLocal = resultFootprintsInit
```
Comment (philvarner) [inline src/components/Search/Search.js:277]:
```suggestion
    const resultFootPrintsLocal = resultFootprintsInit ? resultFootprintsInit : resultFootprints
```

File: src/components/Search/Search.js [hunk 2]
```diff
@@ -389,9 +426,6 @@ const Search = () => {
       >
         <CollectionDropdown error={collectionError}></CollectionDropdown>
       </div>
-      <button className="searchButton" onClick={() => searchAPI()}>
```
Comment (philvarner) [inline src/components/Search/Search.js:392]: is there css for this button that also needs to be removed?

File: src/components/Search/Search.js [hunk 3]
```diff
@@ -223,46 +246,58 @@ const Search = () => {
...
+      }
+      clearTimeout(timeout)
+      timeout = setTimeout(later, wait)
+    }
+  }
+
+  // search throttle set to 1000ms
+  const processSearch = debounce(function (resultFootprintsInit) {
+    searchAPI(resultFootprintsInit)
+  }, 1000)
```
Comment (philvarner) [inline src/components/Search/Search.js:271]: I think the lambda definition here is unnecessary and you can just do:

```suggestion
  const processSearch = debounce(searchAPI, 1000)
```

File: src/components/Search/Search.js [hunk 4]
```diff
@@ -223,46 +246,58 @@ const Search = () => {
...
+      resultFootPrintsLocal.removeLayer(layer)
     })
   }
 
-  // function called when search button clicked
-  function searchAPI() {
-    // get viewport bounds
-    const viewportBounds = map.getBounds()
+  // throttle function to prevent map from rendering too quickly
+  const debounce = (func, wait) => {
```
Comment (philvarner) [inline src/components/Search/Search.js:256]: I like to name variables with the units so it's clear, e.g., `waitInMillis`

File: src/components/Search/Search.js [hunk 5]
```diff
@@ -107,34 +107,57 @@ const Search = () => {
...
+    if (zoomLevelNotice >= MIN_ZOOM) {
       dispatch(setShowZoomNotice(false))
     } else {
       dispatch(setShowZoomNotice(true))
     }
-  }, [zoomLevel])
+  }, [zoomLevelNotice])
 
-  // when dataTime changes, set in global store
+  // when dataTime changes, set in global store and perform new search
```
Comment (philvarner) [inline src/components/Search/Search.js:142]:
```suggestion
  // when datetime changes, set in global store and perform new search
```

**Review by philvarner — CHANGES_REQUESTED** [review]
few comments

File: src/components/Search/Search.js (hunk 1 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:277]: This has been resolved. ESLint suggested this instead:     
`const resultFootPrintsLocal = resultFootprintsInit || resultFootprints
`

File: src/components/Search/Search.js (hunk 2 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:392]: The unused CSS has been removed for this and the BBOX button.

File: src/components/Search/Search.js (hunk 3 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:271]: This has been changed to: 
`const processSearch = (resultFootprintsInit) => debounce(searchAPI(resultFootprintsInit), 1000)`

File: src/components/Search/Search.js (hunk 4 above)
Comment (lucyhutcheson) [reply src/components/Search/Search.js:256]: This has been updated.

**Review by philvarner — APPROVED** [review]

**Pull request merged**